contracts/products/SoteriaCoverageProduct.sol

```diff
@@ -0,0 +1,681 @@
+     * @param paused_ True to pause, false to unpause.
+    */
+    function setPaused(bool paused_) external override onlyGovernance {
+        _paused = paused_;
+        emit PauseSet(paused_);
+    }
+
+    /**
+     * @notice Sets the cooldown period that a user must wait after deactivating their policy, to withdraw funds from their Soteria account.
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @param cooldownPeriod_ Cooldown period in seconds.
+     */
+    function setCooldownPeriod(uint256 cooldownPeriod_) external override onlyGovernance {
+        _cooldownPeriod = cooldownPeriod_;
+        emit CooldownPeriodSet(cooldownPeriod_);
+    }
+
+    /**
+     * @notice set _maxRateNum.
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @param maxRateNum_ Desired maxRateNum.
+    */
+    function setMaxRateNum(uint256 maxRateNum_) external override onlyGovernance {
+        _maxRateNum = maxRateNum_;
+        emit MaxRateNumSet(maxRateNum_);
+    }
+
+    /**
+     * @notice set _maxRateDenom.
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @param maxRateDenom_ Desired maxRateDenom.
+    */
+    function setMaxRateDenom(uint256 maxRateDenom_) external override onlyGovernance {
+        _maxRateDenom = maxRateDenom_;
+        emit MaxRateDenomSet(maxRateDenom_);
+    }
+
+    /**
+     * @notice set _chargeCycle.
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @param chargeCycle_ Desired chargeCycle.
+    */
+    function setChargeCycle(uint256 chargeCycle_) external override onlyGovernance {
+        _chargeCycle = chargeCycle_;
+        emit ChargeCycleSet(chargeCycle_);
+    }
+
+    /**
+     * @notice set _referralRewardPercentage
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @param referralRewardPercentage_ Desired referralRewardPercentage.
+    */
+    function setReferralRewardPercentage(uint256 referralRewardPercentage_) external override onlyGovernance {
+        require(referralRewardPercentage_ <= 10000, "cannot set over 100%");
+        _referralRewardPercentage = referralRewardPercentage_;
+        emit ReferralRewardPercentageSet(referralRewardPercentage_);
+    }
+
+    /***************************************
+    COVER PROMOTION ADMIN FUNCTIONS
+    ***************************************/
+
+    /**
+     * @notice Enables cover promotion admin to gift (and remove) 'free' cover to specific addresses.
+     * Can only be called by the current cover promotion admin.
+     * @param policyholder_ The policy holder to set reward points for.
+     * @param rewardPoints_ Desired amount of reward points.
+    */
+    function setRewardPoints(address policyholder_, uint256 rewardPoints_) external override {
+        require(msg.sender == _registry.get("coverPromotionAdmin"), "not cover promotion admin");
```

Comment:
Do we need a new role in the registry? Has it been updated? I also noticed that we use that same call to registry in the constructor and chargePremiums() function

Reply:
I see, that sounds nice, thank you for clarification